qiskit/circuit/bit.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Quantum bit and Classical bit objects.
+"""
+
+from qiskit.exceptions import QiskitError
+
+
+class Bit:
+    """Implement a generic bit."""
+
+    def __init__(self, register, index):
+        """Create a new generic bit.
+        """
+        try:
+            size = int(index)
+        except Exception:
+            raise QiskitError("index needs to be castable to an int")
+        if size < 0:
+            raise QiskitError("index size must be positive")
+
+        self.register = register
+        self.index = index
+
+    def __repr__(self):
+        """Return the official string representing the bit."""
+        return "%s(%s, %s)" % (self.__class__.__name__, self.register, self.index)
+
+    def __hash__(self):
+        return hash((self.register, self.index))
+
+    def __eq__(self, other):
+        if isinstance(other, tuple):
+            return other[1] == self.index and other[0] == self.register
+        return other.index == self.index and other.register == self.register
+
+    @classmethod
+    def from_tuple(cls, tuple_or_list):
```

Comment:
Can you update the documentation to include the list case? If I have a tuple, I could just call `Bit(*tuple)`  but for a list this function saves writing the list comprehension.

Reply:
Also, do we know that users are always using tuples, and not 2-element lists?

Reply:
This will generate a bit with type determined by which class the `from_tuple` method was called. Do you think there's a use case for `(reg, idx)` => `Qubit` or `Clbit` depending on the type of the register?

Reply:
I just removed the method in 5f44f9b2. Much cleaner...